File: Java/814.java
__________________________________________________________________________________________________
0ms
class Solution {
    public TreeNode pruneTree(TreeNode root) {
        if(root==null) return null;
        if( pruneTree(root.left)==null) root.left=null;
        if( pruneTree(root.right)==null) root.right=null;   
        return (root.right==null && root.left==null && root.val==0) ? null : root;        
    }
}
__________________________________________________________________________________________________
sample 35584 kb submission
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
class Solution {
    public TreeNode pruneTree(TreeNode root) {
        if (root == null) return null;
        root.left = pruneTree(root.left);
        root.right = pruneTree(root.right);
        if (root.val == 0 && root.left == null && root.right == null) return null;
        return root;
    }
}
__________________________________________________________________________________________________

File: Python3/814.py
__________________________________________________________________________________________________
sample 24 ms submission
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def pruneTree(self, root: TreeNode) -> TreeNode:
        def f(tree):
            if tree:
                if tree.left: tree.left = f(tree.left)
                if tree.right: tree.right = f(tree.right)
                if tree.val != 0 or tree.left or tree.right:
                    return tree
        return f(root)
__________________________________________________________________________________________________
sample 13056 kb submission
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def pruneTree(self, root: TreeNode) -> TreeNode:
        if root is None:
            return None

        if root.val == 0 and root.left is None and root.right is None:
            return None
        
        root.left = self.pruneTree(root.left)
        root.right = self.pruneTree(root.right)
        
        if root.val == 0 and root.left is None and root.right is None:
            return None
        
        return root
__________________________________________________________________________________________________
